client/allocrunner/consul_hook.go

```diff
@@ -99,18 +100,15 @@ func (h *consulHook) Prerun() error {
 
 func (h *consulHook) prepareConsulTokensForTask(job *structs.Job, task *structs.Task, tgName string, tokens map[string]map[string]string) error {
 	// if UseIdentity is unset of set to false, quit
-	// FIXME Fetch from Task.Consul.Cluster once #18557 is in
-	consulConfig := h.consulConfigs[structs.ConsulDefaultCluster]
+	consulConfig := h.consulConfigs[task.Consul.Cluster]
 	if consulConfig.UseIdentity == nil || !*consulConfig.UseIdentity {
 		return nil
 	}
 
-	expectedIdentity := task.MakeUniqueIdentityName(tgName)
-
 	// get tokens for alt identities for Consul
 	mErr := multierror.Error{}
 	for _, i := range task.Identities {
-		if i.Name != expectedIdentity {
+		if i.Name != fmt.Sprintf("%s/%s", nomad.ConsulTaskIdentityNamePrefix, task.GetConsulTaskName()) {
```

Comment:
```suggestion
		if i.Name != fmt.Sprintf("%s_%s", nomad.ConsulTaskIdentityNamePrefix, task.GetConsulTaskName()) {
```

I remembered why I switched to not use `/` in the name. We write the signed identity to disk and the name is added to the path. A `/` would cause it to be a subdirectory in Unix systems:
https://github.com/hashicorp/nomad/blob/cecd9b04729e63839428cfa67fdc983ed2182481/nomad/structs/workload_id.go#L42-L44